src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -1376,14 +1386,40 @@ private bool PublishCulture(IContent persistentContent, IEnumerable<ContentVaria
         /// Adds a generic culture error for use in displaying the culture validation error in the save/publish/etc... dialogs
         /// </summary>
         /// <param name="culture">Culture to assign the error to</param>
+        /// <param name="segment">Segment to assign the error to</param>
         /// <param name="localizationKey"></param>
         /// <param name="cultureToken">
         /// The culture used in the localization message, null by default which means <see cref="culture"/> will be used. 
         /// </param>
-        private void AddCultureValidationError(string culture, string localizationKey, string cultureToken = null)
+        private void AddVariantValidationError(string culture, string segment, string localizationKey, string cultureToken = null)
         {
-            var errMsg = Services.TextService.Localize(localizationKey, new[] { cultureToken == null ? _allLangs.Value[culture].CultureName : _allLangs.Value[cultureToken].CultureName });
-            ModelState.AddCultureValidationError(culture, errMsg);
+            var cultureToUse = cultureToken ?? culture;
+            var variantName = GetVariantName(cultureToUse, segment);
+
+            var errMsg = Services.TextService.Localize(localizationKey, new[] { variantName });
+
+            ModelState.AddVariantValidationError(culture, segment, errMsg);
+        }
+
+        /// <summary>
+        /// Creates the human readable variant name based on culture and segment                
+        /// </summary>
+        /// <param name="culture">Culture</param>
+        /// <param name="segment">Segment</param>
+        /// <returns></returns>
+        private string GetVariantName(string culture, string segment)
+        {
+            if(culture.IsNullOrWhiteSpace() && segment.IsNullOrWhiteSpace())
+            {
+                // TODO: Get name for default variant from somewhere?
```

Comment:
What do we expect to happen here?

Reply:
I would propose we return a translated/localized name here.